angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete.component.ts

```diff
@@ -0,0 +1,395 @@
+  highlightListItem(listItemElement): void {
+    if (listItemElement) {
+      listItemElement.isSelected = true;
+
+      // set aria-activedescendant on the input to the id of the
+      // highlighted item
+      if (this.input) {
+        this.renderer.setAttribute(
+          this.input.ref.nativeElement,
+          'aria-activedescendant',
+          listItemElement.ref.nativeElement.id,
+        );
+      }
+    }
+  }
+
+  /**
+   * Emit the itemSelectedEvent with the currently-highlighted Autocomplete
+   * result item.
+   */
+  selectHighlightedListItem(): void {
+    let selectedElement = this.resultItems.filter(
+      (element) => element.isSelected,
+    )[0];
+    let selectedId = selectedElement.ref.nativeElement.id;
+    this.itemSelectedEvent.emit(selectedId);
+  }
+
+  /**
+   * Hide the Autocomplete results list.
+   */
+  hideResults(): void {
+    if (this.results) {
+      // Remove the hidden style
+      this.renderer.addClass(
+        this.results.nativeElement,
+        'sprk-c-Autocomplete__results--hidden',
+      );
+
+      if (this.input) {
+        // Set aria-expanded on the input's parent
+        this.renderer.setAttribute(
+          this.input.ref.nativeElement.parentNode,
+          'aria-expanded',
+          'false',
+        );
+
+        // Remove aria-activedescendant from the input
+        this.renderer.removeAttribute(
+          this.input.ref.nativeElement,
+          'aria-activedescendant',
+        );
+      }
+
+      // remove aria-selected and highlight class from all list items
+      this.removeAllHighlights();
+
+      this.closedEvent.emit();
+      this.isOpen = false;
+    }
+  }
+
+  /**
+   * Show the Autocomplete results list.
+   */
+  showResults(): void {
+    if (this.results) {
+      this.renderer.removeClass(
+        this.results.nativeElement,
+        'sprk-c-Autocomplete__results--hidden',
+      );
+
+      if (this.input) {
+        // Set aria-expanded on the input's parent
+        this.renderer.setAttribute(
+          this.input.ref.nativeElement.parentNode,
+          'aria-expanded',
+          'true',
+        );
+      }
+
+      this.openedEvent.emit();
+      this.isOpen = true;
+    }
+  }
+
+  /**
+   * @ignore
+   * Track the index of the currently-highlighted Autocomplete result item.
+   */
+  highlightedIndex = -1;
+
+  /**
+   * @ignore
+   */
+  ngAfterContentInit(): void {
+    if (this.results) {
+      if (this.isOpen) {
+        this.showResults();
+      } else {
+        this.hideResults();
+      }
+    }
+
+    // if itemSelectedEvent is specified, also set that as the click event on each result
+    if (this.itemSelectedEvent) {
+      this.resultItems.forEach((element) => {
+        element.clickedEvent = this.itemSelectedEvent;
+      });
+    }
+  }
+
+  /**
+   * @ignore
+   * This logic needs to happen in this event so that the correct DOM
+   * attributes will have been set.
+   */
+  ngAfterViewInit(): void {
+    if (this.input) {
```

Comment:
We have two checks here for if there's an input and none of this code runs if there isn't one. What about if we added an early return here at the top instead. it would be like ` if (!this.input) { return; }`.  Then you could reduce the number of if statements and it would be less complex. It would read as "if there isn't an input then don't run any of this code."

Reply:
I changed a couple other spots as well.